pilot/pkg/xds/internalgen.go

```diff
@@ -142,14 +155,111 @@ func (sg *InternalGen) startPush(typeURL string, data []proto.Message) {
 // We can also expose ACKS.
 func (sg *InternalGen) Generate(proxy *model.Proxy, push *model.PushContext, w *model.WatchedResource, updates model.XdsUpdates) model.Resources {
 	res := []*any.Any{}
-	switch w.TypeUrl {
+
+	// Look for ?query parameters.  We don't use url.Parse() because we have no scheme
+	typeURL := w.TypeUrl
+	var qparams url.Values
+	if qIndex := strings.Index(w.TypeUrl, "?"); qIndex >= 0 {
+		typeURL = w.TypeUrl[:qIndex]
+		var err error
+		qparams, err = url.ParseQuery(w.TypeUrl[qIndex+1:])
+		if err != nil {
+			adsLog.Infof("Invalid TypeUrl query params: %q", w.TypeUrl)
+		}
+	}
+
+	switch typeURL {
 	case TypeURLConnections:
 		sg.Server.adsClientsMutex.RLock()
 		// Create a temp map to avoid locking the add/remove
 		for _, v := range sg.Server.adsClients {
 			res = append(res, util.MessageToAny(v.xdsNode))
 		}
 		sg.Server.adsClientsMutex.RUnlock()
+	case TypeDebugSyncronization:
+		res = sg.debugSyncz()
+	case TypeDebugConfigDump:
+		proxyID := qparams.Get("proxyID")
```

Comment:
this should be a resource name, not a query param

Reply:
Fixed.